### ISSUE TITLE
Authorization via Environment Variables?

### ISSUE THREAD

**Issue by infinity-plus** [jeremyephron/simplegmail#49]
Please add support for authorizing Gmail Object via environmental variables for services like Heroku that recommend using Twelve-Factor App principles, e.g., storing config data in the environment instead of files.

**Comment by Tylerlhess** [comment]
While trying to figure out a good way to Dockerize a simple cronjob that checks email to import some stats (its a work around for their API giving only root access to the WAF.) I was going to go down a path of trying to store the creds in a DB. oauth2client does have a SQLacademy storage module but implementing that and pushing it through seems harder than just fixing this 3 years after it was requested and using Docker secrets to store the refresh token and credentials.

TL;DR; might submit a pr for this soon.

**Comment by jeremyephron** [comment]
This is included in [simplegmail 5.0.0](https://github.com/jeremyephron/simplegmail/releases/tag/v5.0.0). Sorry for the long delay, I didn't even realized this was still widely used.

Upgrade with:

```console
python -m pip install --upgrade simplegmail==5.0.0
```

You can now inject existing `google-auth` credentials directly, so credentials can come from an environment variable, secret manager,
database, or another application without creating a token file:

```python
gmail = Gmail(credentials=credentials)
```

There's an example in the README.

**Comment by Tylerlhess** [comment]
No  Issues I am a proponent of OSS so as many of my corporate needs as are reasonable to push back up stream I do. It's also fun to discover I randomly have one off commits to software I'm not using professionally years after submitting them from a different job.
